### PR TITLE
Dashboard component: prepare support for scheduled emails

### PR DESCRIPTION
**Pull request by kandl**
- Copy necessary components to sdk-ui-dashboard package
- Add required selectors and hook for custom implementations
- Add command and event for creating scheduled email
- Align alert commands/events naming with other command and events
- Add hook for easy creation of local component "progress" indicators

JIRA: RAIL-3469

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -287,6 +305,7 @@ const DashboardInner: React.FC<IDashboardProps> = (props: IDashboardProps) => {
     return (
         <InternalIntlWrapper locale={locale}>
             <FilterBarComponent filters={filters} onFilterChanged={onFilterChanged} />
+            {/* <ScheduledEmailDialog /> TODO: cover visibility handling by TopBar button */}
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/dashboard/Dashboard.tsx:308]: Should we leave this in?

File: libs/sdk-ui-dashboard/src/model/commands/scheduledEmail.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+export interface CreateScheduledEmail extends IDashboardCommand {
+    readonly type: "GDC.DASH/CMD.SCHEDULED_EMAIL.CREATE";
+    readonly payload: {
+        readonly scheduledEmail: IScheduledMailDefinition;
+        readonly filterContext?: IFilterContextDefinition;
+    };
+}
+
+/**
+ * Creates the SaveAlert command. Dispatching this command will result in the creating Kpi alert on the backend.
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/commands/scheduledEmail.ts:20]: Mismatching TSDoc.

File: libs/sdk-ui-dashboard/src/dashboard/Dashboard.tsx (hunk 1 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/src/dashboard/Dashboard.tsx]: Removed

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/model/commands/scheduledEmail.ts (hunk 2 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/src/model/commands/scheduledEmail.ts]: Good catch, corrected.

File: libs/sdk-ui-dashboard/src/model/commands/scheduledEmail.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        readonly scheduledEmail: IScheduledMailDefinition;
+        readonly filterContext?: IFilterContextDefinition;
+    };
+}
+
+/**
+ * Creates the CreateScheduledEmail command. Dispatching this command will result in the creating scheduled email on the backend.
+ *
+ * @param scheduledEmail - specify scheduled email to create.
+ * @param filterContext - specify filter context to use for the scheduled email
```
Comment (lupko) [inline libs/sdk-ui-dashboard/src/model/commands/scheduledEmail.ts:23]: please clarify what context will be used if none is provided

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/model/commands/scheduledEmail.ts (hunk 3 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/src/model/commands/scheduledEmail.ts]: Updated.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4034/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5669/ : SUCCESS in 15m 23s

**Review by lupko — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1447

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1174/ : SUCCESS in 1m 04s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1448/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/194/ : SUCCESS in 14m 40s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5671/ : SUCCESS in 15m 43s

**Pull request merged**